src/stream/src/from_proto/source/trad_source.rs

```diff
@@ -190,20 +191,15 @@ impl ExecutorBuilder for SourceExecutorBuilder {
                     state_table_handler,
                 );
 
-                let is_fs_connector = source.with_properties.is_legacy_fs_connector();
+                let is_legacy_fs_connector = source.with_properties.is_legacy_fs_connector();
                 let is_fs_v2_connector = source.with_properties.is_new_fs_connector();
 
-                if is_fs_connector {
-                    #[expect(deprecated)]
-                    crate::executor::source::LegacyFsSourceExecutor::new(
-                        params.actor_context.clone(),
-                        stream_source_core,
-                        params.executor_stats,
-                        barrier_receiver,
-                        system_params,
-                        source.rate_limit,
-                    )?
-                    .boxed()
+                if is_legacy_fs_connector {
+                    // Changed to default since v2.0 https://github.com/risingwavelabs/risingwave/pull/17963
+                    bail!(
+                        "legacy s3 connector is fully deprecated since v2.4.0, please DROP and recreate the s3 source.\nexecutor: {:?}",
+                        params
+                    );
```

Comment:
Will this block the recovery process of the whole cluster? Just to confirm, is there any way for users to drop the source in this situation?

(similar to https://github.com/risingwavelabs/risingwave/issues/17807)

Reply:
IIRC failed recovery won't block DROP. @yezizp2012

Reply:
No, it won't block the DROP command. User can still run drop commands during the recovery.